tests/array.c

```diff
@@ -0,0 +1,199 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  CFL
+ *  ===
+ *  Copyright (C) 2023 The CFL Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <cfl/cfl.h>
+#include "cfl_tests_internal.h"
+
+static void test_create_destroy()
+{
+    struct cfl_array *array = NULL;
+
+    array = cfl_array_create(10);
+    if (!TEST_CHECK(array != NULL)) {
+        TEST_MSG("cfl_array_create failed");
+        return;
+    }
+
+    cfl_array_destroy(array);
+}
+
+static void test_append_fetch()
+{
+    int i;
+    int ret;
+    size_t array_size = 10;
+    struct cfl_array *array = NULL;
+    struct cfl_variant *val = NULL;
+
+    array = cfl_array_create(array_size);
+    if (!TEST_CHECK(array != NULL)) {
+        TEST_MSG("cfl_array_create failed");
+        return;
+    }
+
+    for (i=0; i<(int)array_size; i++) {
```

Comment:
Instead of casting `array_size` both times just define `i` as `size_t` and please rename it as `index`.

Reply:
`size_t` is "unsigned" value on Ubuntu 22.04 and it depends on environment.
I think it needs casting to pass `cfl_array_append_int64` as arg.
```c
#include <stdio.h>

void print_size_t(size_t i)
{
  printf("i=%d\n", i);
}

int main(void) {
  int i = 123;

  print_size_t(i);
  return 0;
}
```

```
$ gcc -Wall a.c
a.c: In function ‘print_size_t’:
a.c:5:14: warning: format ‘%d’ expects argument of type ‘int’, but argument 2 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=]
    5 |   printf("i=%d\n", i);
      |             ~^     ~
      |              |     |
      |              int   size_t {aka long unsigned int}
      |             %ld
```

Reply:
You can use `ssize_t` which is signed for both of them.

Reply:
I modified to use ssize_t.

Reply:
Oh, ssize_t is missing on Windows...
https://ci.appveyor.com/project/fluent/cfl/builds/47128714/job/xq4witur9oh3l6md#L120

Reply:
I fixed ssize_t error on Windows.